bin/MixinGenerator.php

```diff
@@ -0,0 +1,237 @@
+            foreach ($values as $value) {
+                $parts = $this->splitDocLine($value);
+                $type = $parts[0];
+
+                if ($key === 'psalm-assert') {
+                    $type = sprintf($typeTemplate, $type);
+                }
+
+                $comment = sprintf('@%s %s', $key, $type);
+                if (count($parts) >= 2) {
+                    $comment .= sprintf(' %s', implode(' ', array_slice($parts, 1)));
+                }
+
+                $phpdocLines[] = trim($comment);
+            }
+        }
+
+        return $this->staticFunction($newMethodName, $parameters, $parametersDefaults, $phpdocLines, $indent);
+    }
+
+    private function staticFunction(string $name, array $parameters, array $defaultValues, array $phpdocLines, int $indent): string
+    {
+        $indentation = str_repeat(' ', $indent);
+
+        $staticFunction = $this->phpdoc($phpdocLines, $indent) . "\n";
+
+        $staticFunction .= $indentation . 'public static function ' . $name . '(';
+
+        foreach ($parameters as $i => $parameter) {
+            if ($i > 0) {
+                $staticFunction .= ', ';
+            }
+
+            $staticFunction .= '$' . $parameter;
+
+            if (array_key_exists($parameter, $defaultValues)) {
+                $defaultValue = $defaultValues[$parameter] === null ? 'null' : var_export($defaultValues[$parameter], true);
+
+                $staticFunction .= ' = ' . $defaultValue;
+            }
+        }
+
+        $staticFunction .= ');';
+
+        return $staticFunction;
+    }
+
+    private function phpdoc(array $lines, int $indent): string
+    {
+        $indentation = str_repeat(' ', $indent);
+
+        $phpdoc = $indentation . '/**';
+
+        foreach ($lines as $line) {
+            $phpdoc .= "\n" . $indentation . ' * ' . $line;
+        }
+
+        $phpdoc .= "\n" . $indentation . ' */';
+
+        return $phpdoc;
+    }
+
+    private function parseDocComment(string $comment): array
+    {
+        $lines = explode("\n", $comment);
+
+        $result = [];
+
+        foreach ($lines as $line) {
+            if (preg_match('~^\*\s+@(\S+)(\s+.*)?$~', trim($line), $matches)) {
+                if (count($matches) === 2 || $matches[2] === null) {
+                    $matches[2] = '';
+                }
+                $result[$matches[1]][] = trim($matches[2]);
+            }
+        }
+
+        return $result;
+    }
+
+    private function splitDocLine(string $line): array
+    {
+        if (!preg_match('~^(.*)\s+(\$\S+)$~', $line, $matches)) {
+            return [$line];
+        }
+
+        return [trim($matches[1]), $matches[2]];
+    }
+
+    private function getSupportedMethods(ReflectionClass $class): array
+    {
+        $comment = $class->getDocComment();
+
+        preg_match_all('~@method static void ([^(]+)~', $comment, $matches);
+
+        $temporaryUnsupported = [
+            'nullOrNotInstanceOf',
```

Comment:
if these are tracked as psalm-specific issues, let's add links to them

Reply:
That's what I'm doing at this very second yep